### PR TITLE
Improve LS Completions

### LINKED ISSUE #29088: Due to removal to error<*> support LS needs to implement auto-completion logic
**Description:**
Related to PR: https://github.com/ballerina-platform/ballerina-lang/pull/28737  for Ballerina Swan Lake - Beta

These tests are skipped

"error_typedesc1.json"
"error_typedesc2.json"
"error_typedesc3.json"
"error_typedesc4.json"
"error_typedesc5.json"
"error_typedesc6.json"

**Describe your task(s)**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### LINKED ISSUE #29927: Add completion support for xmlns-decl context
**Description:**
$title for both module level xml namespace declaration as well as xmlns declaration statement

**Affected Versions:**
sl-alpha4 at least

### PR DESCRIPTION
**Pull request by nadeeshaan**
## Purpose
Fixes #29088
Fixes #29927

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ModuleXMLNamespaceDeclarationNodeContext.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                && (asKeyword.isEmpty() || asKeyword.get().textRange().startOffset() > cursor)
+                && (namespaceuri.isMissing() || cursor < namespaceuri.textRange().endOffset() + 1);
+    }
+
+    private boolean onSuggestAsKeyword(BallerinaCompletionContext context, ModuleXMLNamespaceDeclarationNode node) {
+        int cursor = context.getCursorPositionInTree();
+        ExpressionNode namespaceuri = node.namespaceuri();
+        Optional<Token> asKeyword = node.asKeyword();
+
+        return !namespaceuri.isMissing() && cursor >= namespaceuri.textRange().endOffset() + 1
```
Comment (IMS94) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ModuleXMLNamespaceDeclarationNodeContext.java:99]: We can rewrite the 2nd condition like this?
```suggestion
        return !namespaceuri.isMissing() && cursor > namespaceuri.textRange().endOffset()
```

**Review by IMS94 — DISMISSED** [review]
Approved with a minor comment.

**Review by pubudu91 — DISMISSED** [review]

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ArrayTypeDescriptorNodeContext.java [hunk 2]
```diff
@@ -72,7 +72,7 @@ public ArrayTypeDescriptorNodeContext() {
 
     private Predicate<Symbol> constantFilter() {
         return symbol -> {
-            if (symbol.kind() != SymbolKind.CONSTANT && symbol.kind() != SymbolKind.ENUM_MEMBER) {
+            if (symbol.kind() != SymbolKind.CONSTANT) {
```
Comment (pubudu91) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ArrayTypeDescriptorNodeContext.java:75]: Btw, why skip enum members for this? 🤔

Comment (nadeeshaan) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ArrayTypeDescriptorNodeContext.java:75]: This is for the array length hence avoid the enums

**Review by mohanvive — APPROVED** [review]

**Pull request merged**